src/audio/module_adapter/module_adapter.c

```diff
@@ -107,12 +107,21 @@ static void module_adapter_mem_free(struct processing_module *mod)
  *	 the create method.
  */
 struct comp_dev *module_adapter_new_ext(const struct comp_driver *drv,
-					const struct comp_ipc_config *config, const void *spec,
+					const struct comp_ipc_config *config,
+					const void *const_spec,
 					void *mod_priv)
 {
 	int ret;
 	struct module_config *dst;
 	const struct module_interface *const interface = drv->adapter_ops;
+	/*
+	 * The const_spec is only couple of stack frames up in the stack so
+	 * casting would be an option too, and it would save couple of
+	 * words from stack,
+	 */
+	struct ipc_config_process spec =
+		*((struct ipc_config_process *) const_spec);
```

Comment:
please use `const` for type-casting just for the peace of mind of imperfect humans like me :-)